ProcessLib/HeatTransportBHE/LocalAssemblers/HeatTransportBHELocalAssemblerSoil-impl.h

```diff
@@ -99,13 +105,13 @@ void HeatTransportBHELocalAssemblerSoil<ShapeFunction, IntegrationMethod>::
     {
         pos.setIntegrationPoint(ip);
         auto& ip_data = _ip_data[ip];
-        auto const& M_ip = ip_data.NTN_product_times_w;
-        auto const& K_ip = ip_data.dNdxTdNdx_product_times_w;
+        auto const& N = ip_data.N;
+        auto const& dNdx = ip_data.dNdx;
+        auto const& w = ip_data.integration_weight;
 
-        auto const k_s =
-            solid_phase
-                .property(
-                    MaterialPropertyLib::PropertyType::thermal_conductivity)
+        // for now only using the solid and liquid phase parameters
+        auto const density_s =
```

Comment:
The variable array `vars` defined in line 99 is never filled with values. But it is used in the entire integration loop for the calculation of properties. If there are non constant constitutive relationships there should be errors.

Reply:
Correct. Now `vars` has been filled with the soil temperature of the integration point. As a result, properties can be temperature-dependent.